geopandas/geoseries.py

```diff
@@ -5,6 +5,7 @@
 import pandas as pd
 from pandas import Series, MultiIndex
 from pandas.core.internals import SingleBlockManager
+from pandas.core.common import flatten
```

Comment:
I know we use `SingleBlockManager` from `core` here but `pandas.core` is generally considered private. I'd rather avoid depending on it if possible. Can we do the fix without `pandas.core.common.flatten`?

Reply:
@martinfleis Thanks for the feedback. Solve the issue using simple Python, your comment also made me realised that  using `pandas.core.common.flatten` was flawed, it will fail when the elements of the MultiIndex were iterables.